blackjax/mcmc/proposal.py

```diff
@@ -90,6 +91,40 @@ def update(initial_energy: float, state: IntegratorState) -> Tuple[Proposal, boo
     return new, update
 
 
+def asymmetric_proposal_generator(
+    energy: Callable,
+    proposal_logdensity_fn,
+    divergence_threshold: float,
+) -> tuple[Callable, Callable]:
+    new, symmetric_update = proposal_generator(energy, divergence_threshold)
+
+    def update(
+        initial_energy: float, state
+    ) -> tuple[Proposal, bool]:
+        """Generate a new proposal from a trajectory state,
+        correcting for asymmetry in the proposal distribution
+        """
+        new_proposal, is_transition_divergent = symmetric_update(initial_energy, state)
+        new_position = new_proposal.state.position
+        previous_position = state.position
+        weight_correction = proposal_logdensity_fn(
```

Comment:
The weight correction can be done directly on the "energy" calculation. Thus, we only need one `proposal_generator` which can do Metropolis, Metropolis Hastings, or any acceptance probability really.